src/lib/list/selection-list.ts

```diff
@@ -479,6 +477,24 @@ export class MatSelectionList extends _MatSelectionListMixinBase implements Focu
     }
   }
 
+  /**
+   * Sets the selected state on all of the options
+   * and emits an event if anything changed.
+   */
+  private _toggleAll(isSelected: boolean) {
+    let hasChanged = false;
```

Comment:
Add a comment like
```ts
// Keep track of whether anything changed because we only want to emit
// a change event if something actually changed.
```